server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1715,6 +1718,46 @@ public RecoveryState recoveryState() {
         return this.recoveryState;
     }
 
+    @Override
+    public ShardLongFieldRange getTimestampMillisRange() {
+        assert isReadAllowed();
+
+        if (mapperService() == null) {
+            return ShardLongFieldRange.MUTABLE; // no mapper service, no idea if the field even exists
+        }
+        final MappedFieldType mappedFieldType = mapperService().fieldType(DataStream.TimestampField.FIXED_TIMESTAMP_FIELD);
+        final boolean hasMappedTimestampField = mappedFieldType instanceof DateFieldMapper.DateFieldType;
+
+        final Engine engine = getEngine();
+        final ShardLongFieldRange rawTimestampFieldRange;
+        try {
+            rawTimestampFieldRange = engine.getRawFieldRange(DataStream.TimestampField.FIXED_TIMESTAMP_FIELD, hasMappedTimestampField);
+        } catch (IOException e) {
+            logger.debug("exception obtaining range for timestamp field", e);
+            return ShardLongFieldRange.MUTABLE;
+        }
+        if (rawTimestampFieldRange == ShardLongFieldRange.MUTABLE) {
+            return ShardLongFieldRange.MUTABLE;
+        }
+        if (rawTimestampFieldRange == ShardLongFieldRange.EMPTY) {
+            return ShardLongFieldRange.EMPTY;
+        }
+
+        if (hasMappedTimestampField == false) {
+            return ShardLongFieldRange.EMPTY;
```

Comment:
My expectation was that if we have a field called `@timestamp` that isn't mapped as a date, and the query contains a date-range filter on the `@timestamp` field, then this shard simply doesn't match. @fcofdez since you're thinking about the logic on the coordinating node, what would you prefer here?

Reply:
I think it would make sense to return a `MUTABLE` instance, and rely on the actual mappings to decide wether or not to apply the filtering?

Reply:
Ok, done in https://github.com/elastic/elasticsearch/pull/65564/commits/c0b0ea373b142a03796db0bdc852d19bffe307d7.

Reply:
The implicit thinking behind making it MUTABLE here is that it is the better way to inform the search part that this is not a date field. This would at least avoid the coordinator having to figure out if a range query for `@timestamp` targets a date field or not. Initially, we intended no mapping info to be available on the coordinator making this important. It still feels more right to me, since we then avoid checking if the mapping is a date mapping on coordinator.

Reply:
Eh actually it's never `false` by the time we get here so the point is moot. I switched it for an assertion in 04bd7541a4a.

Reply:
ahh, but then the interpretation of the `isMappedField` flag in `ReadOnlyEngine` is imprecise, since it assumes that it is not mapped at all, while it could be mapped just not as a date. I wonder if we should just skip calling the engine if it is mapped as anything but a date, returning UNKNOWN in that case?

Reply:
Ah, good point, this is the actual location of that logic. Yes, having it return `UNKNOWN` there in any case without even hitting the engine is fine, see https://github.com/elastic/elasticsearch/pull/65564/commits/c4b497e0a241429082f5ac43bf0b8e5487ffe50b.